programs/lz4io.c

```diff
@@ -539,8 +541,11 @@ static int LZ4IO_compressFilename_extRess(cRess_t ress, const char* srcFileName,
             if (sizeCheck!=outSize) EXM_THROW(35, "Write error : cannot write compressed block");
 
             /* Read next block */
-            readSize  = fread(srcBuffer, (size_t)1, (size_t)blockSize, srcFile);
+            readSize  = fread(srcBuffer, (size_t)1, (size_t)changeBlock, srcFile);
             filesize += readSize;
+
+	    i+=10;
+	    changeBlock=(size_t)LZ4IO_GetBlockSize_FromBlockId (readSize+i);
```

Comment:
`LZ4IO_GetBlockSize_FromBlockId()` is used during decompression to get size of next compressed block,

but here this code lies within `LZ4IO_compressFilename_extRess()`